packages/xhr-http-handler/src/xhr-http-handler.ts

```diff
@@ -0,0 +1,222 @@
+import { HttpHandler, HttpRequest, HttpResponse } from "@aws-sdk/protocol-http";
+import { buildQueryString } from "@aws-sdk/querystring-builder";
+import { HttpHandlerOptions, Provider } from "@aws-sdk/types";
+import { EventEmitter } from "events";
+
+import { requestTimeout } from "./request-timeout";
+
+/**
+ * Represents the http options that can be passed to the xhr http client.
+ */
+export interface XhrHttpHandlerOptions {
+  /**
+   * The number of milliseconds a request can take before being automatically
+   * terminated.
+   */
+  requestTimeout?: number;
+
+  /**
+   * Optionally way to provide instances of xhr.
+   * The default is `new XMLHttpRequest()` from the global scope.
+   * This can be used to attach listeners or other customization.
+   */
+  xhrFactory?: () => XMLHttpRequest | Promise<XMLHttpRequest>;
```

Comment:
Is there any feature depend on this interface? If not, let remove it from the current feature.